include/common/Driver.hpp

```diff
@@ -277,18 +277,25 @@ class Driver
 		_sim->setSectionTimes(secTimes, secCont);
 
 		// Specify initial values
-		if (pp.exists("INIT_STATE_Y") && pp.exists("INIT_STATE_YDOT"))
+		if (pp.exists("INIT_STATE_Y"))
 		{
 			const std::vector<double> initY = pp.getDoubleArray("INIT_STATE_Y");
-			const std::vector<double> initYdot = pp.getDoubleArray("INIT_STATE_YDOT");
-			if (initY.size() != _sim->numDofs() || initY.size() != 2*_sim->numDofs())
-				throw InvalidParameterException("Length of INIT_STATE_Y should be equal to NDOF or 2*NDOF");
-			else if (initY.size() >= _sim->numDofs())
+			if (initY.size() != _sim->numDofs())
 			{
-				if (initYdot.size() >= _sim->numDofs())
-					_sim->applyInitialCondition(initY.data(), initYdot.data());
-				else
-					_sim->applyInitialCondition(initY.data());
+				throw InvalidParameterException("Length of INIT_STATE_Y should be equal to NDOF");
+			}
+			if (pp.exists("INIT_STATE_YDOT"))
+			{
+				const std::vector<double> initYdot = pp.getDoubleArray("INIT_STATE_YDOT");
+				if (initYdot.size() != _sim->numDofs())
+				{
+					throw InvalidParameterException("Length of INIT_STATE_YDOT should be equal to NDOF");
+				}
+				_sim->applyInitialCondition(initY.data(), initYdot.data());
+			}
+			else
+			{
+				_sim->applyInitialCondition(initY.data());
 			}
```

Comment:
@sleweke should we also raise an error if INIT_STATE_YDOT is provided, but no INIT_STATE?

Reply:
You can raise an error or simply ignore it. Up to now, I've opted to simply ignore such data. Maybe emit a warning?

Reply:
That's something we were also discussion on Monday. We should do more error checking on all the inputs. Sometimes it's hard to guess where things went wrong...